src/coreclr/jit/lclvars.cpp

```diff
@@ -2595,25 +2595,35 @@ void Compiler::lvaSetVarLiveInOutOfHandler(unsigned varNum)
 void Compiler::lvaSetVarDoNotEnregister(unsigned varNum DEBUGARG(DoNotEnregisterReason reason))
 {
     noway_assert(varNum < lvaCount);
-    LclVarDsc* varDsc         = &lvaTable[varNum];
+    LclVarDsc* varDsc = &lvaTable[varNum];
+    if (varDsc->lvDoNotEnregister == 1)
+    {
+        return;
+    }
```

Comment:
How about adding something to JitDump if the new reason is not the same as the existing reason?

Reply:
not sure about it, for example, lets say a var is marked as DoNotEnreg 2 times because of `addrExposed` and 2 times because of `lclFld`, the first reason is `addrExposed`, what would you prefer to see it the dump:
1.
```
should not be enregistered because: addrExposed
should not be enregistered also because: lclFld// a new reason
should not be enregistered because: addrExposed // the first reason called again
should not be enregistered also because: lclFld// a new reason, don't have information that we already had lclFld before.
```

2.
```
should not be enregistered because: addrExposed
should not be enregistered also because: lclFld
should not be enregistered also because: addrExposed
should not be enregistered also because: lclFld
```

3.
```
should not be enregistered because: addrExposed
should not be enregistered also because: lclFld
should not be enregistered also because: lclFld
```

Reply:
I was thinking of (3). I think it's better for the `lclFld` lines (in this example) to be duplicated than for there to be no output at all. Unless you tell me the number of duplicated lines will be large (>20), at which point maybe there's some other issue (like, why do we call do-not-enreg so much?)